app/routes.py

```diff
@@ -1,2 +1,312 @@
-from flask import Blueprint
+from app import db
+from app.models.task import Task
+from app.models.goal import Goal
+from flask import request, Blueprint, jsonify 
+from datetime import datetime
+import requests
+import os
+
+
+# class Task:
+#     def __init__(self, id, title, description, completed_at = True):
+#         self.id = id
+#         self.title = title
+#         self.description = description
+#         self.completed_at = completed_at 
+
+
+
+# tasks = [Task(1,"A Brand New Task", "Test Description", False )
+
+tasks_bp = Blueprint("tasks_bp", __name__, url_prefix="/tasks") #functions that handles the endpoint
+goals_bp = Blueprint("goals_bp", __name__, url_prefix="/goals")
+
+
+@tasks_bp.route("", methods = ["POST", "GET"])
+def handle_tasks():
+
+#Wave 1 and Wave 2 
+
+    if request.method == "POST":
+        request_body = request.get_json() #request.get_json makes the text readable in json 
+        if "title" not in request_body or "description" not in request_body or "completed_at" not in request_body:
+            return jsonify({"details":"Invalid data"}), 400
+        
+        new_task = Task(
+            title= request_body["title"],
+            description=request_body["description"],
+            completed_at=request_body["completed_at"])
+
+        db.session.add(new_task)
+        db.session.commit()
+        new_task_response = {"task":{
+                "id": new_task.id,
+                "title" : new_task.title,
+                "description": new_task.description,
+                "is_complete": False if new_task.completed_at == None else True 
+                }
+        }
```

Comment:
Once you implement the instance method in your task model, you can just call the instance method here instead of writing all of this code over and over.